scripts/skills/perks/perk_rf_ghostlike.nut

```diff
@@ -13,24 +18,73 @@ this.perk_rf_ghostlike <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
+	function isHidden()
+	{
+		return this.m.Enemies.len() == 0 && !this.getContainer().getActor().getCurrentProperties().IsImmuneToZoneOfControl;
+	}
+
 	function getTooltip()
 	{
-		local tooltip = this.skill.getTooltip();
+		local ret = this.skill.getTooltip();
+
+		if (this.getContainer().getActor().getCurrentProperties().IsImmuneToZoneOfControl)
+		{
+			ret.push({
+				id = 6,
+				type = "text",
+				icon = "ui/icons/special.png",
+				text = ::Reforged.Mod.Tooltips.parseString("The next movement will ignore [Zone of Control|Concept.ZoneOfControl]")
+			});
+		}
 
-		tooltip.push({
-			id = 6,
-			type = "text",
-			icon = "ui/icons/special.png",
-			text = ::Reforged.Mod.Tooltips.parseString("The next movement will ignore [Zone of Control|Concept.ZoneOfControl]")
-		});
+		if (this.m.Enemies.len() != 0)
+		{
+			ret.push({
+				id = 6,
+				type = "text",
+				icon = "ui/icons/special.png",
+				text = ::MSU.Text.colorGreen("25%") + " increased damage and " + ::MSU.Text.colorGreen("+20%") + " damage ignoring armor against:"
+			});
 
-		return tooltip;
+			foreach (enemy in this.m.Enemies)
+			{
+				local enemy = ::Tactical.getEntityByID(enemy);
+				if (enemy == null || !enemy.isPlacedOnMap() || !enemy.isAlive())
+					continue;
+
+				ret.push({
+					id = 10,
+					type = "text",
+					icon = "ui/orientation/" + enemy.getOverlayImage() + ".png",
+					text = enemy.getName()
+				});
+			}
+
+			ret.push({
+				id = 10,
+				type = "text",
+				icon = "ui/icons/warning.png",
+				text = ::MSU.Text.colorRed("The damage bonus will be lost upon swapping an item or waiting or ending your turn")
+			});
+		}
+
+		return ret;
 	}
 
-	function onUpdate( _properties )
+	function isEnabled()
 	{
-		this.m.IsHidden = true;
+		if (this.m.IsForceEnabled)
+			return true;
+
+		if (this.getContainer().getActor().getItems().getStaminaModifier([::Const.ItemSlot.Body, ::Const.ItemSlot.Head]) >= this.m.ArmorStaminaModifier)
```

Comment:
You are comparing negative numbers here so you need to invert the comparison operator to get the result you want.

At this point I propose changing the member variable `this.m.ArmorStaminaModifier` into `this.m.MaximumWeight` and making it into a positive number.
And then here you can write
```
local currentWeight = -1 * this.getContainer().getActor().getItems().getStaminaModifier([::Const.ItemSlot.Body, ::Const.ItemSlot.Head]);
if (currentWeight > this.m.MaximumWeight)
	return false;
```
 That will improve readability/parsability; at least I find working with positive weight numbers more intuitive than negative staminamodifier numbers.